src/runtime_src/core/edge/drm/zocl/zocl_xclbin.c

```diff
@@ -840,6 +840,19 @@ zocl_xclbin_read_axlf(struct drm_zocl_dev *zdev, struct drm_zocl_axlf *axlf_obj,
 			DRM_INFO("disable partial bitstream download, "
 			    "axlf flags is %d", axlf_obj->za_flags);
 		} else {
+			 /*
+			  * cleanup previously loaded xclbin related data
+			  * before loading new bitstream/pdi
+			  * */
+			if (kds_mode == 1 && (zocl_xclbin_get_uuid(zdev) != NULL)) {
+				subdev_destroy_cu(zdev);
+				if (zdev->aie) {
+					zocl_aie_reset(zdev);
+					mutex_lock(&zdev->aie_lock);
+					zdev->aie->aie_reset = false;
```

Comment:
This is a bit weird because we just reset AIE but saying aie_reset is false. I think we are safe to move aie out of reset state only after the PDI is successfully loaded.

There might be a corner case if we don't set the aie_reset correctly, e.g. somehow the following load process fails but we record aie is not reset. Then if user load a non-dfx xclbin, and then access AIE array, this will cause SError.

Reply:
This is because of following check in zocl_create_aie. It fails to create aie if aie_reset is true
https://github.com/Xilinx/XRT/blob/master/src/runtime_src/core/edge/drm/zocl/zocl_aie.c#L242
I think, best is to call zocl_destory_aie instead of setting this variable.
https://github.com/Xilinx/XRT/blob/master/src/runtime_src/core/edge/drm/zocl/zocl_aie.c#L319

Reply:
I understand why you are setting it. My point is we need a mechanism to protect AIE array from accessing before a PDI is loaded, using this flag or something else. 
Setting this flag to false before loading PDI will compromise this protection.

Reply:
Let me set it after successful pdi load. Hope that should be fine.

Reply:
Can't we check the presence AIE_METADAT section for that purpose?

Reply:
Then we are safe to set aie_reset to true after loading PDI, right?
But I am totally fine with your proposal to destroy aie. And seems to me, this destroy aie operation should be done before loading pdi.